libnavigation-base/src/main/java/com/mapbox/navigation/base/route/Router.kt

```diff
@@ -1,9 +1,22 @@
 package com.mapbox.navigation.base.route
 
 import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.route.model.Route
 
 interface Router {
 
-    fun getRoute(origin: Point, waypoints: List<Point>, callback: (route: Route) -> Unit)
+    fun getRoute(
+        origin: Point,
+        waypoints: List<Point> = emptyList(),
+        destination: Point,
+        callback: Callback
+    )
+
     fun cancel()
+
+    interface Callback {
+        fun onRouteReady(routes: List<Route>)
```

Comment:
What about `onResponse` instead? `onRouteReady` doesn't reflect that we're returning a list of `Route`s.